lib/stb_image.ex

```diff
@@ -103,6 +105,204 @@ defmodule StbImage do
         "unsupported tensor shape: #{inspect(shape)} (expected height-width-channel)"
       )
 
+  if Code.ensure_loaded?(Kino.Render) do
+    defimpl Kino.Render do
+      require Logger
+
+      @default_kino_render_encoding Application.compile_env(
+                                      :stb_image,
+                                      :kino_render_encoding,
+                                      :png
+                                    )
+      @doc """
+      Get preferred image encoding when rendering in Kino.
+
+      Default value is `Application.compile_env(:stb_image, :kino_render_encoding, :png)`.
+      """
+      @spec kino_render_encoding(atom()) :: term()
+      def kino_render_encoding(encoding \\ nil)
+
+      def kino_render_encoding(nil) do
+        Process.get(:stb_image_kino_render_encoding, @default_kino_render_encoding)
+      end
+
+      def kino_render_encoding(encoding) when encoding in [:png, :jpg, :jpeg] do
+        encoding =
+          if encoding == :jpeg do
+            :jpg
+          else
+            encoding
+          end
+
+        Process.put(:stb_image_kino_render_encoding, encoding)
+      end
+
+      def kino_render_encoding(unsupported_encoding) do
+        raise RuntimeError, "Encoding #{inspect(unsupported_encoding)} is not supported."
+      end
```

Comment:
Sorry, to be clear, I think you can validate those at compile time:

```suggestion
      render_encoding = Application.compile_env(:stb_image, :kino_render_encoding, :png)
      
      if render_encoding not in [:png, :jpg, :jpeg] do
        raise "invalid value for :kino_render_encoding. Expected one of :png, :jpg, :jpeg, got #{inspect(render_encoding)}"
      end

      @default_kino_render_encoding render_encoding
```

And the same for the other ones. :)

Now you don't need the function, you can use the module attribute directly below.

Reply:
Oh I was thinking that maybe it could useful if the user can change the behaviour. For the example, when they are using Livebook for some tutorials, for some sections they can set the default tab to Image while using Numerical tab for some other sections. 

Also, having these functions could be easier for beginners to play around because the atoms `:stb_image_kino_render_*` in `Process.put(:stb_image_kino_render_*, value)` do not get any autocompletions etc.

What do you think?

Reply:
As for the encoding part, switching between png and jpg encoding might help? As png gives the exact image while jpg can reduce the bandwidth.

Or perhaps we can add another function like `display(image, encoding)`…?

Reply:
If that's the case, I don't think we should have both the Application environment and the process dictionary, instead, we simply rely on the application environment. The user can change it as:

    Application.fetch_env!(:stb_image, :kino_render_encoding, :jpg)

Make sure to add:

```
def application do
  [env: [kino_render_encoding: :png, ...]]
end
```

with the default values.

We could also add a `display` function in the future, but we also want to make it easier to render images directly from Kino too, so I would wait a bit. :)

Reply:
Got it! I’ll apply the suggestion. ;)

Reply:
We may even end-up with a Kino.Image protocol, so we can easily convert anything to image. We will see :)

Reply:
Just to make sure we are on the same page: @cocoa-xu, could you please meanwhile change this to use Application.fetch_env! at runtime instead of the process dictionary? Thank you!

Reply:
Yes, I will change this too :)

Reply:
Okay I applied the suggested changes :D